code/modules/power/supermatter/crystal_matrix.dm

```diff
@@ -0,0 +1,176 @@
+GLOBAL_LIST_EMPTY(crystal_matrix)
+
+/obj/destabilized_supermatter
+	name = "destabilized crystal"
+	desc = "A strangely translucent and iridescent crystal."
+	icon = 'icons/obj/supermatter.dmi'
+	icon_state = "psy"
+	density = TRUE
+	anchored = TRUE
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1
+	light_range = 4
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+/obj/destabilized_supermatter/Initialize()
+	. = ..()
+	end_world_activation()
+	SSshuttle.registerHostileEnvironment(src)
+
+/obj/destabilized_supermatter/Destroy()
+	return..()
+
+///Start the process where the crystal matrix expands to devour the world
+/obj/destabilized_supermatter/proc/end_world_activation()
+	priority_announce("WARNING - Possible TK-end of the world scenario approaching, please standby")
```

Comment:
* stand by

Reply:
aight, I'm not gonna grammar check the rest of this PR, but suffice it to say that it does indeed need a proofreading by someone

Reply:
yeah there are errors, english is not my first language so i make a lot of gramatical errors